sdk/keyvault/Azure.Security.KeyVault.Keys/tests/KeyClientTests.cs

```diff
@@ -222,6 +223,42 @@ public void ReleaseKeyParameterValidation()
             Assert.AreEqual("target", ex.ParamName);
         }
 
+        [Test]
+        [SyncOnly]
+        public void GetCryptographyClientValidation()
+        {
+            ArgumentException ex = Assert.Throws<ArgumentNullException>(() => Client.GetCryptographyClient(null));
+            Assert.AreEqual("name", ex.ParamName);
+
+            ex = Assert.Throws<ArgumentException>(() => Client.GetCryptographyClient(string.Empty));
+            Assert.AreEqual("name", ex.ParamName);
+        }
+
+        [Test]
+        public async Task GetCryptographyClientUsesSamePipeline()
```

Comment:
Do we want to test when giving a key/version arg which doesn't exists on the Vault?

What about testing non-url-friendly key/version args like `key?Name` or `version=1`  .  Would the URI creation fail and bobble up the exception ?

Reply:
We don't validate arguments apart from null (and possibly empty) checks based on all languages' guidelines I've seen. The service should fail accordingly, and we'll throw an exception as a result.